src/utils/vc/credentials.js

```diff
@@ -466,22 +472,200 @@ export async function issueCredential(
+  }
+  // eslint-disable-next-line no-nested-ternary
+  const key = full ? (credential.credentialSchema.fullJsonSchema !== undefined ? 'fullJsonSchema' : 'id') : 'id';
+  return CredentialSchema.convertFromDataUri(credential.credentialSchema[key]);
+}
+
+/**
+ * Get status of a credential issued with revocation type of private status list.
+ * @param credential
+ * @returns {*|Object|undefined}
+ */
+export function getPrivateStatus(credential) {
+  return credential.credentialStatus && credential.credentialStatus.type === PrivateStatusList2021EntryType ? credential.credentialStatus : undefined;
+}
+
+/**
+ * Verify the credential status given the private status list credential
+ * @param credentialStatus - `credentialStatus` field of the credential
+ * @param privateStatusListCredential
+ * @param documentLoader
+ * @param suite
+ * @param verifyStatusListCredential - Whether to verify the status list credential. This isn't necessary when the caller
+ * of this function got the credential directly from the issuer.
+ * @param expectedIssuer - Checks whether the issuer of the private status list credential matches the given
+ * @returns {Promise<{verified: boolean}>}
+ */
+export async function verifyPrivateStatus(credentialStatus, privateStatusListCredential, {
+  documentLoader = null, suite = [], verifyStatusListCredential = true, expectedIssuer = null,
+}) {
+  const fullSuite = [
+    new Ed25519Signature2018(),
+    new EcdsaSepc256k1Signature2019(),
+    new Sr25519Signature2020(),
+    new JsonWebSignature2020(),
+    ...suite,
+  ];
+  const { statusPurpose: credentialStatusPurpose } = credentialStatus;
+  const { statusPurpose: slCredentialStatusPurpose } = privateStatusListCredential.credentialSubject;
+  if (slCredentialStatusPurpose !== credentialStatusPurpose) {
+    throw new Error(
+      `The status purpose "${slCredentialStatusPurpose}" of the status list credential does not match the status purpose "${credentialStatusPurpose}" in the credential.`,
+    );
+  }
+
+  // ensure that the issuer of the verifiable credential matches
+  if (typeof expectedIssuer === 'string') {
+    const issuer = typeof privateStatusListCredential.issuer === 'object' ? privateStatusListCredential.issuer.id : privateStatusListCredential.issuer;
+    if (issuer !== expectedIssuer) {
+      throw new Error(`Expected issuer to be ${expectedIssuer} but found ${issuer}`);
+    }
+  }
+
+  // verify VC
+  if (verifyStatusListCredential) {
+    const verifyResult = await verifyCredential(privateStatusListCredential, {
+      suite: fullSuite,
+      documentLoader,
+    });
+    if (!verifyResult.verified) {
+      throw new Error(`Status list credential failed to verify with error: ${verifyResult.error}`);
+    }
+  }
+
+  // decode list from SL VC
+  // get JSON StatusList
+  const { credentialSubject: sl } = privateStatusListCredential;
+  const { encodedList } = sl;
+  const list = await decodeList({ encodedList });
```

Comment:
Wouldn't be better to accept `privateStatusListCredential` as an instance of `PrivateStatusList2021Credential` and use `await privateStatusListCredential.decodedStatusList()` instead?

Reply:
Yes. I had missed the caching aspect. Fixed now. Thanks